tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs

```diff
@@ -161,61 +196,164 @@ internal Configuration BuildXmlConfiguration(IEnumerable<Type> cmdletTypList)
+            public  string ScriptBlock { get; set; }
+            public ViewControl TargetView { get; set; }
+        }
+
         internal void HandleMemberInfo<T>(
             IEnumerable<T> memberInfoList,
             IList<TableColumnHeader> tableHeaders, 
             IList<TableColumnItem> tableColumnItems, 
-            IList<ListItem> listItems) 
+            IList<ListItem> listItems,
+            ref GroupByInfo groupByInfo) 
             where T: MemberInfo
         {
-
             var memeber = typeof(T) == typeof(PropertyInfo)
                 ? "property"
                 : "field";
 
             var memberInfoListFiltered = memberInfoList
                 .Where(p => {
                     if (!_onlyMarkedProperties) return true;
-                    var ps1XmlAttribute = (Ps1XmlAttribute)Attribute.GetCustomAttribute(p, typeof(Ps1XmlAttribute));
-                    return ps1XmlAttribute != null;
+                    var ps1XmlAttributes = Attribute.GetCustomAttributes(p, typeof(Ps1XmlAttribute));
+                    return ps1XmlAttributes.Length > 0;
                 })
-                .OrderBy(p => p.Name)
                 .ToList();
 
             if (memberInfoListFiltered.Count == 0) return;
 
+            var columnHeadersWithPosition = new List<Tuple<TableColumnHeader, uint>>();
+            var columnItemsWithPosition = new List<Tuple<TableColumnItem, uint>>();
+            var listItemsWithPosition = new List<Tuple<ListItem, uint>>();
+
             foreach (var memberInfo in memberInfoListFiltered)
             {
-                var ps1XmlAttribute = _onlyMarkedProperties
-                    ? (Ps1XmlAttribute)Attribute.GetCustomAttribute(memberInfo, typeof(Ps1XmlAttribute))
-                    : null;
+                if (_onlyMarkedProperties)
+                {
+                    var ps1XmlAttributes = Attribute.GetCustomAttributes(memberInfo, typeof(Ps1XmlAttribute));
+                    foreach (var attribute in ps1XmlAttributes)
+                    {
+                        var ps1XmlAttribute = (Ps1XmlAttribute) attribute;
+                        if (ps1XmlAttribute.GroupByThis)
+                        {
+                            if (groupByInfo == null)
+                            {
+                                groupByInfo = new GroupByInfo
+                                {
+                                    Label = ps1XmlAttribute.Label,
+                                    ScriptBlock = ps1XmlAttribute.ScriptBlock,
+                                    PropertyName = memberInfo.Name,
+                                    TargetView = ps1XmlAttribute.Target,
+                                };
+
+                                continue;
+                            }
+                            else
+                            {
+                                // duplicate!
+                                // FIXME: replace or ignore or throw?
```

Comment:
whats going to happen in this case? currently we seem to be only using the first group by we see.. the later ones will be ignored. We should leave th e user with a good default behavior.
I think we can even throw an exception and say that only one Group By is allowed.